sdk/core/Azure.Core/tests/ManagementRecordedTestBaseTests.cs

```diff
@@ -203,5 +203,37 @@ public void ValidateForwardClientCallDefault()
             var ex = Assert.ThrowsAsync<InvalidOperationException>(async () => testResource = await testResource.GetForwardsCallDefaultAsync());
             Assert.AreEqual(ex.Message, "Expected some diagnostic scopes to be created, found none");
         }
+
+        [TestCase(RecordedTestMode.Record)]
+        [TestCase(RecordedTestMode.Playback)]
+        [TestCase(RecordedTestMode.Live)]
+        public async Task ValidateWaitOverride(RecordedTestMode mode)
+        {
+            // keep the curent test mode and restore it back when finished, otherwise it will invoke unnecessary clean-up
+            var currentMode = Mode;
+
+            Mode = mode;
+            try
+            {
+                ManagementTestClient testClient = InstrumentClient(new ManagementTestClient());
+                TestResource testResource = testClient.GetTestResource();
+                Stopwatch sw = Stopwatch.StartNew();
+                testResource = (await testResource.GetLroAsync(WaitUntil.Completed)).Value;
+                sw.Stop();
+                Assert.AreEqual("TestResourceProxy", testResource.GetType().Name);
+                if (mode == RecordedTestMode.Playback)
+                {
+                    Assert.LessOrEqual(sw.ElapsedMilliseconds, 1000);
+                }
+                else
+                {
+                    Assert.GreaterOrEqual(sw.ElapsedMilliseconds, 1000);
```

Comment:
The test failed on CI with the following error message. I think it's a noise, so we could add some tolerance.
> Error Message:
     Expected: greater than or equal to 1000
  But was:  999

```suggestion
                    Assert.That(sw.ElapsedMilliseconds, Is.GreaterThanOrEqualTo(1000).Within(50));
```